lib/components/primitive-components/Trace/Trace.ts

```
@@ -781,12 +780,102 @@ export class Trace
+        toPortName = this.props.to
+      }
+
+      if (!fromPort.source_port_id) {
+        throw new Error(
+          `Missing source_port_id for the 'from' port (${fromPortName}).`,
+        )
+      }
+
+      // Handle `from` port label
+      const existingFromNetLabel = db.schematic_net_label
+        .list()
+        .find((label) => label.source_net_id === fromPort.source_port_id)
+
+      if (!existingFromNetLabel) {
+        db.schematic_net_label.insert({
+          text: this.props.schDisplayLabel!,
+          source_net_id: fromPort.source_port_id!,
+          anchor_position: fromAnchorPos,
+          center: fromAnchorPos,
+          anchor_side:
+            getEnteringEdgeFromDirection(fromPort.facingDirection!) ?? "bottom",
+        })
+      } else if (existingFromNetLabel.text !== this.props.schDisplayLabel) {
+        throw new Error(
+          `Cannot create net label for port ${fromPortName} because it already has a net label with text "${existingFromNetLabel.text}".`,
+        )
+      }
+
+      // Handle `to` port label
+      const toAnchorPos = portsWithPosition[1].position
+      const toPort = portsWithPosition[1].port
 
-    this.schematic_trace_id = trace.schematic_trace_id
+      if (!toPort.source_port_id) {
+        throw new Error(
+          `Missing source_port_id for the 'to' port (${toPortName}).`,
+        )
+      }
+
+      const existingToNetLabel = db.schematic_net_label
+        .list()
+        .find((label) => label.source_net_id === toPort.source_port_id)
+
+      if (!existingToNetLabel) {
+        db.schematic_net_label.insert({
+          text: this.props.schDisplayLabel!,
+          source_net_id: toPort.source_port_id!,
+          anchor_position: toAnchorPos,
+          center: toAnchorPos,
+          anchor_side:
+            getEnteringEdgeFromDirection(toPort.facingDirection!) ?? "bottom",
+        })
+      } else if (existingToNetLabel.text !== this.props.schDisplayLabel) {
```

hmm interesting, I guess that could be an issue, nice job on catching this edge case, but don't use these big "if elses" when you can just throw an error early

```
if (existingToNetLabel && existingToNetLabel.text !== this.props.schDisplayLabel) {
  throw new Error(...)
}
```

Now you don't need the "else" part!